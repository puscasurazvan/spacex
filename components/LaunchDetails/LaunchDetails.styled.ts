import styled from 'styled-components';

export const List = styled.ul({
  display: 'flex',
  flexDirection: 'column',
  listStyle: 'none',
  '>*': {
    marginTop: '5px',
  },
  overfloWrap: 'break-word',
  wordWrap: 'break-word',
});

export const Item = styled.li({
  fontWeight: 'bolder',
});

export const Span = styled.span({
  fontWeight: 'bolder',
  color: '#8e8e8e',
});
