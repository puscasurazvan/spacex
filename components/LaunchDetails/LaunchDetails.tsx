import * as Styled from './LaunchDetails.styled';
import { v4 as uuid } from 'uuid';

type LaunchDetailsProps = {
  launchDate: string;
  core: string;
  payloadType: string;
  payloadId: string;
  failureReason: string;
};

export const LaunchDetails = ({ launchDate, core, payloadType, payloadId, failureReason }: LaunchDetailsProps) => {
  const launchDetails = [
    {
      testId: 'date',
      spanText: 'DATE:',
      value: launchDate,
    },
    {
      testId: 'core',
      spanText: 'CORE:',
      value: core,
    },
    {
      testId: 'payloadType',
      spanText: 'PAYLOAD TYPE:',
      value: payloadType,
    },
    {
      testId: 'payloadId',
      spanText: 'PAYLOAD ID:',
      value: payloadId,
    },
  ];

  return (
    <Styled.List>
      {launchDetails.map(({ testId, spanText, value }) => (
        <Styled.Item data-testid={testId} key={uuid()}>
          <Styled.Span>{spanText}</Styled.Span> {value}
        </Styled.Item>
      ))}

      {failureReason && (
        <Styled.Item data-testid="failureReason">
          <Styled.Span>FAILURE REASON:</Styled.Span> {failureReason}
        </Styled.Item>
      )}
    </Styled.List>
  );
};
