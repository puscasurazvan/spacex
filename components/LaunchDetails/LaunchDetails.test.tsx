import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LaunchDetails } from './LaunchDetails';

describe('LaunchDetails Component', () => {
  it('renders the LaunchDetails component with all details', () => {
    const launchDate = '2023-07-27';
    const core = 'Falcon 9';
    const payloadType = 'Satellite';
    const payloadId = 'ABCD1234';
    const failureReason = 'Engine anomaly detected';

    const { container, getByTestId } = render(
      <LaunchDetails
        launchDate={launchDate}
        core={core}
        payloadType={payloadType}
        payloadId={payloadId}
        failureReason={failureReason}
      />
    );
    const dateElement = getByTestId('date');
    const coreElement = getByTestId('core');
    const payloadIdElement = getByTestId('payloadId');
    const payloadTypeElement = getByTestId('payloadType');
    const failureReasonElement = getByTestId('failureReason');

    expect(container).toMatchSnapshot();

    expect(dateElement).toBeInTheDocument();
    expect(coreElement).toBeInTheDocument();
    expect(payloadIdElement).toBeInTheDocument();
    expect(payloadTypeElement).toBeInTheDocument();
    expect(failureReasonElement).toBeInTheDocument();
  });

  it('renders the LaunchDetails component without failure reason', () => {
    const launchDate = '2023-07-27';
    const core = 'Falcon 9';
    const payloadType = 'Satellite';
    const payloadId = 'ABCD1234';
    const failureReason = '';

    const { container, getByTestId } = render(
      <LaunchDetails
        launchDate={launchDate}
        core={core}
        payloadType={payloadType}
        payloadId={payloadId}
        failureReason={failureReason}
      />
    );
    const dateElement = getByTestId('date');
    const coreElement = getByTestId('core');
    const payloadIdElement = getByTestId('payloadId');
    const payloadTypeElement = getByTestId('payloadType');

    expect(container).toMatchSnapshot();
    expect(dateElement).toBeInTheDocument();
    expect(coreElement).toBeInTheDocument();
    expect(payloadIdElement).toBeInTheDocument();
    expect(payloadTypeElement).toBeInTheDocument();
  });
});
