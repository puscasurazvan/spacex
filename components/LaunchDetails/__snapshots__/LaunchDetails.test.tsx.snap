// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchDetails Component renders the LaunchDetails component with all details 1`] = `
<div>
  <ul
    class="sc-bczSft gJhYnS"
  >
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="date"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        DATE:
      </span>
       
      2023-07-27
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="core"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        CORE:
      </span>
       
      Falcon 9
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="payloadType"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        PAYLOAD TYPE:
      </span>
       
      Satellite
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="payloadId"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        PAYLOAD ID:
      </span>
       
      ABCD1234
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="failureReason"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        FAILURE REASON:
      </span>
       
      Engine anomaly detected
    </li>
  </ul>
</div>
`;

exports[`LaunchDetails Component renders the LaunchDetails component without failure reason 1`] = `
<div>
  <ul
    class="sc-bczSft gJhYnS"
  >
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="date"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        DATE:
      </span>
       
      2023-07-27
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="core"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        CORE:
      </span>
       
      Falcon 9
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="payloadType"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        PAYLOAD TYPE:
      </span>
       
      Satellite
    </li>
    <li
      class="sc-gsnScm hhaiOH"
      data-testid="payloadId"
    >
      <span
        class="sc-dkzCjb ipmQqA"
      >
        PAYLOAD ID:
      </span>
       
      ABCD1234
    </li>
  </ul>
</div>
`;
